cvat/apps/dataset_manager/bindings.py

```diff
@@ -535,7 +535,13 @@ def _import_shape(self, shape, parent_label_id=None):
                 self.soft_attribute_import and attrib.name not in CVAT_INTERNAL_ATTRIBUTES
             )
         ]
-        _shape['points'] = list(map(float, _shape['points']))
+
+        # TODO: remove once importers are guaranteed to return correct type 
+        # (see https://github.com/cvat-ai/cvat/pull/8226/files#r1695445137)
+        points = _shape["points"]
+        for i, point in enumerate(map(float, points)):
+            points[i] = point
```

Comment:
If the idea is to avoid copying the array, you may just drop using `map`

```
for i, point in enumerate(points):
    points[i] = float(point)
```

Reply:
I know that map does not copy the array, I mean that it is not necessary here as you anyway iterate over the array.

Reply:
As I mentioned above, `map` is still more efficient than suggested changes, justifying its usage. But even if it weren't, I don't really see the reason to remove it in the first place. Could you elaborate on that?

Reply:
If you are looking for best optimizations from interpretator POV, try this one:

```
points[:] = map(float, points)
```

Reply:
This might be faster, but it will not be memory efficient, since interpreter will first have to generate the whole new list before updating the current one, so this would roughly be equivalent of
```py
new_points = list(map(float, points))
points.clear()
points.extend(new_points)
```